### PR TITLE
Add a .NET Core 2.0 Hello World sample

### PR DESCRIPTION
**Pull request by joaoasrosa**
This PR adds a simple `HelloWorld` sample.
It uses Cake as build platform for the AWS Lambda.

It also updates the documentation to reflect the changes.

### CODE REVIEW

File: samples/hello-world/dotnetcore2.0/build.cake [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    .IsDependentOn("Restore")
+    .IsDependentOn("Build")
+    .IsDependentOn("Publish")
+    .IsDependentOn("Pack")
+    .IsDependentOn("Run-Local")
+    .Does(() => { Information("Run target ran."); });
+    
+Task("Default")
+    .Description("This is the default task which will run if no specific target is passed in.")
+    .IsDependentOn("Default");
```
Comment (MattHodge) [inline samples/hello-world/dotnetcore2.0/build.cake]: This causes a `Reflexive edges in graph are not allowed` exception (the task being dependant on itself).

**Comment by joaoasrosa** [comment]
@MattHodge thanks for pointing it out. New commit fixing the issue.

**Comment by ashemedai** [comment]
The refresh messed up this PR now though. :\

**Comment by sanathkr** [comment]
Yes, it messes up the PR but for a good reason :) We added a dotnet sample consumable by `sam init` command - https://github.com/awslabs/aws-sam-cli/tree/develop/samcli/local/init/templates. Can you rebase and merge your changes with this sample and update the PR? 

`sam init` is a new command we added in 0.3.0. It allows customers to quickly get started with a sample app in a runtime of their choice without the need to clone repos from Github. Your PR will be way more valuable, and reach more people, if you merge it with the dotnet template vended by `sam init`

**Comment by joaoasrosa** [comment]
@sanathkr will do. Thanks

**Comment by ashemedai** [comment]
@sanathkr Ah, I see, all the samples are now cookiecutter templates for `init`, correct?

**Comment by sanathkr** [comment]
Precisely!

**Comment by joaoasrosa** [comment]
Closing this PR, in order to preserve the history.
A new PR was open: https://github.com/awslabs/aws-sam-cli/pull/401
Thanks for all the feedback @MattHodge, @ashemedai and @sanathkr. @sanathkr let me know if something is missing.

**Pull request closed without merging**